runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionStateSampler.java

```diff
@@ -117,7 +117,8 @@ public Optional<ActiveMessageMetadata> getActiveMessageMetadataForWorkId(String
     return Optional.ofNullable(null);
   }
 
-  public Map<String, IntSummaryStatistics> getProcessingDistributionsForWorkId(String workId) {
+  public synchronized Map<String, IntSummaryStatistics> getProcessingDistributionsForWorkId(
+      String workId) {
     if (!activeTrackersByWorkId.containsKey(workId)) {
```

Comment:
```suggestion
    // synchronized to avoid race condition of accessing activeTrackersByWorkId 
    if (!activeTrackersByWorkId.containsKey(workId)) {
```